src/main/java/ee/carlrobert/codegpt/completions/ollama/OllamaServerAgent.java

```diff
@@ -0,0 +1,272 @@
+package ee.carlrobert.codegpt.completions.ollama;
+
+import static java.lang.String.format;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessListener;
+import com.intellij.execution.process.ProcessOutputType;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.util.Key;
+import ee.carlrobert.codegpt.completions.ServerAgent;
+import ee.carlrobert.codegpt.completions.llama.LlamaCompletionModel;
+import ee.carlrobert.codegpt.completions.llama.ServerStartupParams;
+import ee.carlrobert.codegpt.settings.service.util.ServerProgressPanel;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Service
+public final class OllamaServerAgent implements ServerAgent {
```

Comment:
I did not yet test the local execution of an Ollama Server. My idea is to tell the user to install ollama locally via https://ollama.ai/download and then we can use the simple CLI to start the server